docs: Use relative path to the assets dir

surajssd opened the pull request:
The docs currently have absolute path which is always erroneous. This
commit changes that to the relative path which can be just copy pasted
and things work fine.

rata commented on docs/flatcar-linux/aws.md:
```diff
@@ -208,7 +208,7 @@ Check the [variables.tf](https://github.com/kinvolk/lokomotive-kubernetes/blob/m
 | dns_zone_id | AWS Route53 DNS zone id | "Z3PAABBCFAKEC0" |
 | ssh_authorized_key | SSH public key for user 'core' | "ssh-rsa AAAAB3NZ..." |
-| asset_dir | Path to a directory where generated assets should be placed (contains secrets, used to destroy the cluster), cannot be a relative path | "/home/user/.secrets/clusters/tempest" |
+| asset_dir | Path to a directory where generated assets should be placed (contains secrets, used to destroy the cluster), cannot be a relative path | "./assets" |
```
It says "cannot be a relative path" :joy: And AWS is the only one that says this?

Maybe this was because of a limitation in other tools using this that are now fixed? :thinking: I'm quite sure it is not a problem in using the terraform module with a relative path, though. So I think it is safe to remove this? Maybe we should double check this, my memory is not the best :)

surajssd replied:
removed it.

Testing it in a bit.

rata replied:
Thanks! Maybe the CI uses a relative path for tests or we can change that? In that way, you don't have to test :)

surajssd replied:
Yes, this good idea :-)

rata commented on docs/flatcar-linux/aws.md:
```diff
@@ -109,7 +109,7 @@ module "aws-tempest" {
   # configuration
   ssh_authorized_key = "ssh-rsa AAAAB3Nz..."
-  asset_dir          = "/home/user/.secrets/clusters/tempest"
+  asset_dir          = "./assets"
```
This will break this line: https://github.com/kinvolk/lokomotive-kubernetes/pull/82/files#diff-d8962d6ba4490833047bc8e884ed9d23R165

Idem for all the other docs

surajssd replied:
nice catch, fixed it.

rata commented on docs/flatcar-linux/aws.md:
```diff
@@ -208,7 +208,7 @@ Check the [variables.tf](https://github.com/kinvolk/lokomotive-kubernetes/blob/m
 | dns_zone_id | AWS Route53 DNS zone id | "Z3PAABBCFAKEC0" |
 | ssh_authorized_key | SSH public key for user 'core' | "ssh-rsa AAAAB3NZ..." |
-| asset_dir | Path to a directory where generated assets should be placed (contains secrets, used to destroy the cluster), cannot be a relative path | "/home/user/.secrets/clusters/tempest" |
+| asset_dir | Path to a directory where generated assets should be placed (contains secrets, used to destroy the cluster). | "./assets" |
```
I will check with git-blame who wrote it can't be a relative path, and ping just in case we are missing something

rata replied:
@surajssd feel free to close this if you think testing is enough instead of pinging :)

pothos replied:
Maybe write, can be relative or absolute but if `~` is included, path expanding via `pathexpand("yourpath")` is needed.

surajssd commented:
So when I changed the CI code from
`asset_dir = "${pathexpand("~/lokomotive-assets")}"` to
`asset_dir = "~/lokomotive-assets"` CI job failed with following error:

```bash
module.aws-tempest.null_resource.bootkube-start: Provisioning with 'file'...

Error: Error applying plan:

1 error occurred:
	* module.aws-tempest.null_resource.bootkube-start: stat /root/lokomotive-assets: no such file or directory



Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

But when I changed it to relative path like following:

`asset_dir = "./lokomotive-assets"`, it works fine. I am not sure what is wrong.

rata commented:
@surajssd we can check the tfstate to see where it was replaced and where it was not. My guess, by doing a quick test (see below) is that `~` is being expanded in some places and not in others in the terraform code.

For example, if you do this:

```
resource "template_dir" "config" {
  # Create a valid `a` dir with some file inside
  source_dir = "./a/"
  destination_dir = "~/non-existant-dir/"
  #destination_dir = "./non-existant-dir/"
}
```

It will not copy it to the home directory, it will create a folder in the current directory named '~'. And as most of our assets are generated with `template_dir` I guess they are not created in the home (`/root/lokomotive-assets`) but, however, some other part of the code might be intepreting `~` as home and using `/root/lokomotive-assets` and that is why I **think** it shows that error:

> 1 error occurred:
>	* module.aws-tempest.null_resource.bootkube-start: stat /root/lokomotive-assets: no such file or directory

But haven't actually checked.

invidian commented:
@surajssd can you please resolve merge conflicts? I'd like to get this PR to the conclusion.
